.buildkite/pipeline.yml

```diff
@@ -119,9 +119,10 @@ steps:
           - universal
           - x64
           - arm64
+        notify:
+          - github_commit_status:
+              context: All Mac Dev Builds
     if: build.tag !~ /^v[0-9]+/
```

Comment:
Unrelated to this PR, but given this `if` is attached to the parent `group` (hence his indentation level) yet is the only additional attribute of this `group` node after the long `steps` one, wdyt about moving it between `key:` and `steps:` on line 78? And while at it add a `# comment` mentioning we only do dev builds on non-tags (and similar for the `release-mac` group wrt release builds only on tags)?

```yaml
    # Dev Builds only if not a git tag
  - group: 📦 Build for Mac
    key: dev-mac
    if: build.tag !~ /^v[0-9]+/
    steps:
      - label: 🔨 Mac Dev Build - {{matrix}}
        agents:
          queue: mac
        command: |
          # long script taking multiple lines…
        plugins: …
        artefact_paths: …
        matrix: …
        notify: …

# …then on lines 176–223:

    # Release Builds only if a git tag
  - group: 📦 Build for Mac
    key: release-mac
    if: build.tag =~ /^v[0-9]+/
    steps:
      - label: 🔨 Mac Release Build - {{matrix}}
        agents:
          queue: mac
        command: |
          # long script taking multiple lines…
        plugins: …
        artefact_paths: …
        matrix: …
        notify: …
```

An even further improvement (?) would be to DRY the group/step for Dev vs Release, e.g. in a `mac-build.yml` pipeline that would use env vars like `$VARIANT` to be parametrized dynamically and account for the small diff between Dev vs Release, and then we'd be able to call `VARIANT=RELEASE buildkite-agent pipeline upload mac-build.yml` to upload the appropriate steps for the right variant depending on if we're using a tag or not…

[Similar to how Beeper is doing it to account for Nightly vs Production variants](https://github.com/beeper/beeper-desktop-new/blob/main/.buildkite/pipelines/official-build.yml#L43-L53) actually.

Reply:
Yes to moving the `if` at the top. The indentation confused me but I didn't make the leap on how to address it. Thanks!

Also yes about DRYing. I thought about it when I saw the duplication, but never had a chance. Should have made an issue for it...